pkg/controller/networkaddonsconfig/networkaddonsconfig_controller.go

```diff
@@ -593,6 +607,26 @@ func isSCCAvailable(c kubernetes.Interface) (bool, error) {
 	return isResourceAvailable(c, "securitycontextconstraints", "security.openshift.io", "v1")
 }
 
+// isMonitoringAvailable checks if we can deploy the monitoring component
+func IsMonitoringAvailable(c kubernetes.Interface) (bool, error) {
+	prometheusRuleResourceAvailable, err := isResourceAvailable(c, "customresourcedefinitions/prometheusrules.monitoring.coreos.com", "apiextensions.k8s.io", "v1")
+	if err != nil {
+		return false, errors.Wrap(err, "failed to check if prometheusRule resource is available")
+	}
+
+	serviceMonitorResourceAvailable, err := isResourceAvailable(c, "customresourcedefinitions/servicemonitors.monitoring.coreos.com", "apiextensions.k8s.io", "v1")
+	if err != nil {
+		return false, errors.Wrap(err, "failed to check if serviceMonitor resource is available")
+	}
+
+	if prometheusRuleResourceAvailable && serviceMonitorResourceAvailable {
+		return true, nil
+	}
+
+	log.Printf("will not deploy monitoring manifests: not all monitoring resources are available: %s: %v, %s, %v", monitoringv1.PrometheusRuleKind, prometheusRuleResourceAvailable, monitoringv1.ServiceMonitorsKind, serviceMonitorResourceAvailable)
```

Comment:
maybe worth to return this error string as err in the next line ?
this way the caller will print it as err, as it does for the other errors above

Reply:
It's kind of semantic, since I'm not really failing CNAO process if `IsMonitoringAvailable()` returns an error.
The difference between this printf and the other errors, is that in this case - nothing really failed. The cluster just doesn't have the right CRDs to run the monitoring instances, so we can't deploy monitoring. 
So instead of returning an error, I'm adding this printf that also sets it on the logs so it will be heard..